api/internal/managers/linux/infra/upgrade.go

```diff
@@ -262,12 +275,37 @@ func (m *infraManager) distributeArtifacts(ctx context.Context, in *ecv1beta1.In
 	channelID := m.releaseData.ChannelRelease.ChannelID
 	appVersion := m.releaseData.ChannelRelease.VersionLabel
 
-	// TODO: configure when we support airgap (note: we'll have to push the LAM image to the EC registry first somehow)
+	// Determine the local artifact mirror image path
 	localArtifactMirrorImage := versions.LocalArtifactMirrorImage
 
+	// For airgap installations, rewrite the LAM image to point to the EC registry
+	if registrySettings != nil && registrySettings.HasLocalRegistry {
+		destImage, err := destECImage(registrySettings, localArtifactMirrorImage)
+		if err != nil {
+			return fmt.Errorf("determine LAM image path in EC registry: %w", err)
+		}
+		localArtifactMirrorImage = destImage
+	}
+
 	return m.upgrader.DistributeArtifacts(ctx, in, localArtifactMirrorImage, license.Spec.LicenseID, appSlug, channelID, appVersion)
 }
 
+// destECImage returns the location to an EC image in the registry
+// reference from KOTS: https://github.com/replicatedhq/kots/blob/d26ebd2acaccc54313e7f7d5ca3ca580ae1a0bc5/pkg/imageutil/image.go#L105
+func destECImage(registrySettings *types.RegistrySettings, srcImage string) (string, error) {
+	// parsing as a docker reference strips the tag if both a tag and a digest are used
+	parsed, err := reference.ParseDockerRef(srcImage)
+	if err != nil {
+		return "", fmt.Errorf("failed to normalize source image %s: %w", srcImage, err)
+	}
+	srcImage = parsed.String()
+
+	imageParts := strings.Split(srcImage, "/")
+	lastPart := imageParts[len(imageParts)-1]
+
+	return filepath.Join(registrySettings.Host, registrySettings.Namespace, "embedded-cluster", lastPart), nil
```

Comment:
is this a file or and image ref? if its an image ref should you use a different library other than filepath?